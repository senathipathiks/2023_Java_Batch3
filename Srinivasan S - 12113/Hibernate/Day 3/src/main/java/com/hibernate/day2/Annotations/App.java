package com.hibernate.day2.Annotations;

import java.util.List;
import java.util.Scanner;

import org.hibernate.Session;
import org.hibernate.SessionFactory;
import org.hibernate.Transaction;
import org.hibernate.cfg.Configuration;

/**
 * Hello world!
 *
 */
public class App 
{
    public static void main( String[] args )
    {
       Configuration cfg=new Configuration().configure("hibernate.cfg.xml");
       SessionFactory factory=cfg.buildSessionFactory();
       Session session= factory.openSession();
       Transaction t=session.beginTransaction();
       
       Employee e1=new Employee();
       
       //insert data
//       e1.seteName("Surya");
//       e1.seteDesig("Devops");
 //      session.persist(e1);
       
//       //update data
//       Employee temp=session.find(Employee.class, 2);
//       temp.seteDesig("Support");
//       session.merge(temp);
       
       //delete data
//       Employee temp=session.find(Employee.class, 5);
//       session.remove(temp);
       
       
//       List<Employee>list=session.createQuery("from Employee").list();
//       for (Employee employee : list) {
//		System.out.println(employee);
//	}
//       System.out.println("Donee");
       
       Scanner sc=new Scanner(System.in);
       System.out.println("Welcome To Sri group of Industries");
       int msg=0;

       do {
		
    	          System.out.println("Choose your menu");
       System.out.println("1.Insert\n2.Update\n3.FindAll\n4.Delete\n5.Find");
       int option=sc.nextInt();
       
       switch (option) {
	case 1:
		
		System.out.println("Enter Employee name ");
		String name=sc.next();
		
		System.out.println("Enter Employee designation");
		String des=sc.next();
		
		e1.seteName(name);
		e1.seteDesig(des);
		session.persist(e1);
		
		t.commit();
		break;
		
	case 2:
		
		System.out.println("Enter Employee id");
		int uid=sc.nextInt(); 
		
		System.out.println("Enter Employee name ");
		String uname=sc.next();
		
		System.out.println("Enter Employee designation");
		String udes=sc.next();
		
		Employee temp=session.find(Employee.class,uid);
		temp.seteName(uname);
		temp.seteDesig(udes);
		
		t.commit();
		break;
		
	case 3:
		List<Employee>list=session.createQuery("from Employee").list();
		
		for (Employee employee : list) {
			System.out.println(employee);
		}
		
		break;
	
	case 4:
		System.out.println("Enter Employee id");
		int did=sc.nextInt(); 
		
		Employee tempp=session.find(Employee.class, did);
		session.remove(tempp);
		t.commit();
		break;
		
	
	case 5:
		System.out.println("Enter Employee id");
		int fid=sc.nextInt();
		
		Employee tempfind=session.find(Employee.class, fid);
		System.out.println(tempfind);
		
	default:
		break;
	}
       System.out.println("Want to continue");
		System.out.println("1.Yes\n2.No");
		 msg=sc.nextInt();
       
    
    } while (msg==1);
       System.out.println("Thank you !");
    }
}
