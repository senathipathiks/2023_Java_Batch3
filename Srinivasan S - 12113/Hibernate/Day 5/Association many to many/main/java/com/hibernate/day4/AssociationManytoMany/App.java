package com.hibernate.day4.AssociationManytoMany;

import org.hibernate.Session;
import org.hibernate.SessionFactory;
import org.hibernate.Transaction;
import org.hibernate.cfg.Configuration;

/**
 * Hello world!
 *
 */
public class App 
{
    public static void main( String[] args )
    {
    	Configuration cfg=new Configuration().configure("hibernate.cfg.xml");
        SessionFactory factory=cfg.buildSessionFactory();
        Session session=factory.openSession();
        Transaction t=session.beginTransaction();
        
        Employee emp1=new Employee();
        emp1.setEroll(113);
        emp1.setName("Sundar");
       
        Project proj1=new Project();
        proj1.setpId(002);
        proj1.setpName("Hospital Management System");
        
     // employee can work on two projects, Add project references in the employee
        emp1.getProjects().add(proj1);
        
     // Add employee reference in the projects
        proj1.getEmp().add(emp1);
        
        session.persist(emp1);
        session.persist(proj1);
        
        t.commit();
        session.close();
        		
    }
}
