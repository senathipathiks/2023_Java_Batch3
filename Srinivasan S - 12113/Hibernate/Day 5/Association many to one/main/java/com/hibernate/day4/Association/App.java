package com.hibernate.day4.Association;

import org.hibernate.Session;
import org.hibernate.SessionFactory;
import org.hibernate.Transaction;
import org.hibernate.cfg.Configuration;

/**
 * Hello world!
 *
 */
public class App 
{
    public static void main( String[] args )
    {
        Configuration cfg=new Configuration().configure("hibernate.cfg.xml");
        SessionFactory factory=cfg.buildSessionFactory();
        Session session=factory.openSession();
        Transaction t=session.beginTransaction();
        
        Department d3=new Department(1113,"Development");
//        Department d4=new Department(1112,"Testing");
        
        Employee emp1=new Employee(10,"Kaviya", d3);
        Employee emp2=new Employee(11,"Sundar",d3);
//        Employee emp3=new Employee(6,"Varshinee",d4);
        session.persist(emp1);
        session.persist(emp2);
//        session.persist(emp3);
         t.commit();
         Employee emp3=new Employee(12,"Kaviya", d3);

         t.begin();
         session.persist(emp3);
         t.commit();
        
    }
}
