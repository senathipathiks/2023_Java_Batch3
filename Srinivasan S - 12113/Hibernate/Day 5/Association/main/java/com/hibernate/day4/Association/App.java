package com.hibernate.day4.Association;

import org.hibernate.Session;
import org.hibernate.SessionFactory;
import org.hibernate.Transaction;
import org.hibernate.cfg.Configuration;

/**
 * Hello world!
 *
 */
public class App 
{
    public static void main( String[] args )
    {
        Configuration cfg=new Configuration().configure("hibernate.cfg.xml");
        SessionFactory factory=cfg.buildSessionFactory();
        Session session=factory.openSession();
        Transaction t=session.beginTransaction();
        
        Laptop l=new  Laptop(111,"Dell");
        session.persist(l);
        
        Employee emp=new Employee(1,"Srini",l);
        session.persist(emp);
        
        
        
        t.commit();
        
    }
}
