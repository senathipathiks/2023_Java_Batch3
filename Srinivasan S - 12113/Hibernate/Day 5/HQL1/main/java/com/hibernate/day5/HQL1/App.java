package com.hibernate.day5.HQL1;

import java.util.Iterator;
import java.util.List;
import java.util.Random;



import org.hibernate.Session;
import org.hibernate.SessionFactory;
import org.hibernate.Transaction;
import org.hibernate.cfg.Configuration;

import jakarta.persistence.Query;

/**
 * Hello world!
 *
 */
public class App 
{
	@SuppressWarnings({"unchecked","deprecation"})
    public static void main( String[] args )
    {
    	Configuration cfg=new Configuration().configure("hibernate.cfg.xml");
        SessionFactory factory=cfg.buildSessionFactory();
        Session session=factory.openSession();
        Transaction t=session.beginTransaction();
        Random r=new Random();
        
//        for (int i = 0; i < 15; i++) {
//			Employee emp=new Employee();
//			emp.seteId(i);
//			emp.seteName("Name"+i);
//			emp.seteDept(r.nextInt(4));
//			session.persist(emp);
//		}
        
//        Query query=session.createQuery("From Employee");
//        List<Employee>list=query.getResultList();
//        for (Employee e : list) {
//			System.out.println(e);
//		}
        
//        Query query=session.createQuery("From Employee where eDept>1");
//        List<Employee>list=query.getResultList();
//        
//        for (Employee e : list) {
//			System.out.println(e);
//		}
        
//        Query query=session.createQuery("From Employee where eId=12");
//        List<Employee>list=query.getResultList();
//        Employee emp=(Employee)((org.hibernate.query.Query) query).uniqueResult();              
//			System.out.println(list);
//			System.out.println(emp);
        
//        Query query=session.createQuery("select eId,eName,eDept From Employee where eId=5");
//    
//      Object[] emp=(Object[])((org.hibernate.query.Query) query).uniqueResult();
//      System.out.println(emp[0]+" : "+emp[1]+" : "+emp[2]);
        
//        Query query=session.createQuery("select count(eId) from Employee where eDept>0");
//      List<Employee> list=query.getResultList();
//      
//      for (Object count : list) {
//			System.out.println(count);
//		}
        
//        Query query=session.createQuery("update Employee set eName=:n where eId=:i");
//        query.setParameter("n", "Srinivasan");
//        query.setParameter("i", 1);
//        int status=query.executeUpdate();
//        
//        if(status>0) {
//        	System.out.println("Success");
//        }
//        else
//        	System.out.println("Failure")

        Query query=session.createQuery("delete Employee where eId=9");
        
        int status=query.executeUpdate();
        if(status>0) {
        	System.out.println("Success");
        }
        else
        	System.out.println("Failure");
        
        t.commit();
    }
}
