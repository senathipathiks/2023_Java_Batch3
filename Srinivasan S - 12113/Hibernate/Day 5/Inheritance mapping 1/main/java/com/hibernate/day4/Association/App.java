package com.hibernate.day4.Association;

import org.hibernate.Session;
import org.hibernate.SessionFactory;
import org.hibernate.Transaction;
import org.hibernate.cfg.Configuration;

/**
 * Hello world!
 *
 */
public class App 
{
    public static void main( String[] args )
    {
        Configuration cfg=new Configuration().configure("hibernate.cfg.xml");
        SessionFactory factory=cfg.buildSessionFactory();
        Session session=factory.openSession();
        Transaction t=session.beginTransaction();
        
       Employee e1=new RegularEmp(1001,"Srinivasan",1000.0f,10000);
       
       Employee e2=new ContractEmp(1002,"Arun",500.0f,"2 years");       
        
      session.save(e1);
      session.save(e2);
        
       t.commit();
        
    }
}
