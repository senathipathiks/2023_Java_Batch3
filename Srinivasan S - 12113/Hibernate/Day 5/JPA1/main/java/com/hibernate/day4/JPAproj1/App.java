package com.hibernate.day4.JPAproj1;

import javax.persistence.EntityManager;
import javax.persistence.Persistence;

/**
 * Hello world!
 *
 */
public class App 
{
    public static void main( String[] args )
    {
       EntityManager emf=Persistence.createEntityManagerFactory("CourseDB").createEntityManager();
       emf.getTransaction().begin();
       Course c1=new Course(103,"C++","CSE");
       emf.persist(c1);
       emf.getTransaction().commit();
       emf.close();
    }
}
