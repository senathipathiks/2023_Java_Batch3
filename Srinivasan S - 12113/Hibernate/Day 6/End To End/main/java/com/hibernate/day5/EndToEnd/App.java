package com.hibernate.day5.EndToEnd;

import org.hibernate.Session;
import org.hibernate.SessionFactory;
import org.hibernate.Transaction;
import org.hibernate.cfg.Configuration;

public class App 
{
    public static void main( String[] args )
    {
    	Configuration cfg=new Configuration().configure();
        SessionFactory factory=cfg.buildSessionFactory();
        Session session=factory.openSession();
        Transaction t=session.beginTransaction();
        
        User u1=new User();
        u1.setUserName("Srinivasan");
        
        Address add1=new Address();
        add1.setuStreet("Melpapanattu");
        add1.setuCity("Neyveli");
        
        
        Mobile m1=new Mobile();
        m1.setBrand("Samsung");
        m1.setModel("Galaxy s24 ultra");
        
        
        Vehicle v1=new Vehicle();
        v1.setvName("Vento");
        
        // user and address
        u1.setAddress(add1);
        add1.setUser(u1);
        
        //user and mobile        
        m1.setUser(u1);
        u1.getMob().add(m1);
        
        //user and vehicle
        v1.getUser().add(u1);
        u1.getVeh().add(v1);
        
        session.persist(u1);
//        session.persist(add1);
//        session.persist(m1);
//        session.persist(v1);
        
        t.commit();
        
        session.close();
        
        
        
    }
}
