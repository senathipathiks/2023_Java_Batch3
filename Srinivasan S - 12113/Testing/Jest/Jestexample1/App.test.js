import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import App from "./App";

describe("Test suite for app", () => {
  // test("renders learn react link", () => {
  //   render(<App />);
  //   const linkElement = screen.getByText(/Welcome !!!/i);
  //   expect(linkElement).toBeInTheDocument();
  // });
  // it.skip("renders Hi Relevantz", () => {
  //   render(<App />);
  //   const linkElement = screen.getByText("Hi Relevantz ");
  //   expect(linkElement).toBeInTheDocument();
  // });

  test("renders snapshot", () => {
    const comp = renderer.create(<App />);
    const tree = comp.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
