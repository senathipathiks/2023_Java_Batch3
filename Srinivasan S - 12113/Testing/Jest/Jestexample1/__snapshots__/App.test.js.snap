// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test suite for app renders snapshot 1`] = `
<div
  className="App"
>
  <h1>
    Welcome !!
  </h1>
  <br />
  <h2>
    Hi Relevantz !!
  </h2>
</div>
`;
